urls.py

```diff
@@ -45,70 +43,14 @@
          name='{}-detail'.format(name))
     for name, dview in detail_views.items()
 )
-
-# Test JS template view
-# TODO: Generalize like above!
 urlpatterns.extend(
-    [
-        path(
-            'popupjs/reg_mun_pop/<int:pk>/',
-            views.RegMunPopDetailJsView.as_view(),
-            name='reg_mun_pop_popupjs'
-        ),
-        path(
-            'popupjs/reg_mun_energy_re_el_dem_share/<int:pk>/',
-            views.RegMunEnergyReElDemShareDetailJsView.as_view(),
-            name='reg_mun_energy_re_el_dem_share_popupjs'
-        ),
-        path(
-            'popupjs/reg_mun_gen_energy_re/<int:pk>/',
-            views.RegMunGenEnergyReDetailJsView.as_view(),
-            name='reg_mun_gen_energy_re_popupjs'
-        ),
-        path(
-            'popupjs/reg_mun_gen_energy_re_per_capita/<int:pk>/',
-            views.RegMunGenEnergyRePerCapitaDetailJsView.as_view(),
-            name='reg_mun_gen_energy_re_per_capita_popupjs'
-        ),
-        path(
-            'popupjs/reg_mun_gen_energy_re_density/<int:pk>/',
-            views.RegMunGenEnergyReDensityDetailJsView.as_view(),
-            name='reg_mun_gen_energy_re_density_popupjs'
-        ),
-        path(
-            'popupjs/reg_mun_gen_cap_re/<int:pk>/',
-            views.RegMunGenCapReDetailJsView.as_view(),
-            name='reg_mun_gen_cap_re_popupjs'
-        ),
-        path(
-            'popupjs/reg_mun_gen_cap_re_density/<int:pk>/',
-            views.RegMunGenCapReDensityDetailJsView.as_view(),
-            name='reg_mun_gen_cap_re_density_popupjs'
-        ),
-        path(
-            'popupjs/reg_mun_dem_el_energy/<int:pk>/',
-            views.RegMunDemElEnergyDetailJsView.as_view(),
-            name='reg_mun_dem_el_energy_popupjs'
-        ),
-        path(
-            'popupjs/reg_mun_dem_el_energy_per_capita/<int:pk>/',
-            views.RegMunDemElEnergyPerCapitaDetailJsView.as_view(),
-            name='reg_mun_dem_el_energy_per_capita_popupjs'
-        ),
-        path(
-            'popupjs/reg_mun_dem_th_energy/<int:pk>/',
-            views.RegMunDemThEnergyDetailJsView.as_view(),
-            name='reg_mun_dem_th_energy_popupjs'
-        ),
-        path(
-            'popupjs/reg_mun_dem_th_energy_per_capita/<int:pk>/',
-            views.RegMunDemThEnergyPerCapitaDetailJsView.as_view(),
-            name='reg_mun_dem_th_energy_per_capita_popupjs'
-        )
-    ]
+    path('popupjs/{}/<int:pk>/'.format(name), dview.as_view(
+        template_name='stemp_abw/popups/js_popup.html'),
+         name='{}-js'.format(name))
+    for name, dview in detail_views.items()
 )
 
-# search JSON data views classes and append to URLs
+# Search JSON data-view-classes and append to URLs
```

Comment:
Notice: In composite nouns in English, you only use hyphens if parts of speech are different (e.g. noun+adverb), but not with 2 or more nouns

Reply:
Yes, there is one hyphen to much, I learned that you use hypens for compound nouns (except for the last word) if you have 3+ words. On the Internet I found concerning hyphens and nouns: ["Rule 2b. When writing out new, original, or unusual compound nouns, writers should hyphenate whenever doing so avoids confusion"](https://www.grammarbook.com/punctuation/hyphens.asp). So I will change this to "data-view classes".